### PR TITLE
[HIGH PRIORITY] abstract entitiy class

### PR DESCRIPTION
**Pull request by IDragonfire**
https://github.com/FAForever/faf-java-api/pull/74

### CODE REVIEW

**Comment by IDragonfire** [comment]
I currently only add a superclass, switching `int` to `Integer` is something bigger: https://github.com/FAForever/faf-java-api/issues/83

File: src/test/java/com/faforever/integration/factories/ClanFactory.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.faforever.integration.factories;
+
+import com.faforever.api.data.domain.Clan;
+import com.faforever.api.data.domain.Player;
+import lombok.Builder;
+
+public class ClanFactory {
+
+  public static Clan create() {
```
Comment (IDragonfire) [inline src/test/java/com/faforever/integration/factories/ClanFactory.java]: I played around with the `@Builder` from lombok and if you must specify more you must call the `builder`

Comment (IDragonfire) [inline src/test/java/com/faforever/integration/factories/ClanFactory.java]: We can rename it maybe to `createDefault`

File: src/main/java/com/faforever/api/data/domain/AbstractEntity.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.MappedSuperclass;
+import java.time.OffsetDateTime;
+
+@MappedSuperclass
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
```
Comment (IDragonfire) [inline src/main/java/com/faforever/api/data/domain/AbstractEntity.java]: `@MappedSuperclass` is for `JPA`
and `@Inhertiance` defines the Strategy and we need `TABLE_PER_CLASS` from my understanding ...
I will test it ...

Comment (IDragonfire) [inline src/main/java/com/faforever/api/data/domain/AbstractEntity.java]: Looks like you are right ...
http://stackoverflow.com/a/9669719

**Comment by micheljung** [comment]
After other PRs are merged and conflicts in this are resolved, I'll merge this.

**Pull request merged**